crates/hyperswitch_connectors/src/connectors/elavon/transformers.rs

```diff
@@ -1,228 +1,587 @@
+}
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PaymentResponse {
+    ssl_issuer_response: Option<SslIssuerResponse>,
+    ssl_result: SslResult,
+    ssl_txn_id: String,
+    ssl_result_message: Option<String>,
 }
 
-impl<F, T> TryFrom<ResponseRouterData<F, ElavonPaymentsResponse, T, PaymentsResponseData>>
-    for RouterData<F, T, PaymentsResponseData>
+impl<F>
+    TryFrom<
+        ResponseRouterData<F, ElavonPaymentsResponse, PaymentsAuthorizeData, PaymentsResponseData>,
+    > for RouterData<F, PaymentsAuthorizeData, PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: ResponseRouterData<F, ElavonPaymentsResponse, T, PaymentsResponseData>,
+        item: ResponseRouterData<
+            F,
+            ElavonPaymentsResponse,
+            PaymentsAuthorizeData,
+            PaymentsResponseData,
+        >,
     ) -> Result<Self, Self::Error> {
-        Ok(Self {
-            status: common_enums::AttemptStatus::from(item.response.status),
-            response: Ok(PaymentsResponseData::TransactionResponse {
-                resource_id: ResponseId::ConnectorTransactionId(item.response.id),
-                redirection_data: Box::new(None),
-                mandate_reference: Box::new(None),
-                connector_metadata: None,
-                network_txn_id: None,
-                connector_response_reference_id: None,
-                incremental_authorization_allowed: None,
-                charge_id: None,
+        let status = enums::AttemptStatus::foreign_from((
+            &item.response,
+            item.data.request.is_auto_capture()?,
+        ));
+
+        let response = match &item.response {
+            ElavonPaymentsResponse::Error(error) => Err(ErrorResponse {
+                code: error.error_code.clone(),
+                message: error.error_message.clone(),
+                reason: Some(error.error_message.clone()),
+                attempt_status: None,
+                connector_transaction_id: None,
+                status_code: item.http_code,
             }),
+            ElavonPaymentsResponse::Success(response) => {
+                if status == enums::AttemptStatus::Failure {
+                    Err(ErrorResponse {
+                        code: response
+                            .ssl_result_message
+                            .clone()
+                            .unwrap_or_else(|| "NO_ERROR_CODE".to_string()),
+                        message: response
+                            .ssl_result_message
+                            .clone()
+                            .unwrap_or_else(|| "NO_ERROR_MESSAGE".to_string()),
+                        reason: Some(
+                            response
+                                .ssl_result_message
+                                .clone()
+                                .unwrap_or_else(|| "NO_ERROR_MESSAGE".to_string()),
+                        ),
```

Comment:
We do not get any error messages indicating why the payment failed?

Reply:
we get only ssl_result_message. I had kept it optional but checked again , its coming for all the transcations
so we can map directly